game/plugins/src/main/kotlin/gg/rsmod/plugins/content/items/amuletofglory/amulet_of_glory.plugin.kts

```diff
@@ -0,0 +1,119 @@
+package gg.rsmod.plugins.content.items.amuletofglory
+
+import gg.rsmod.plugins.content.magic.TeleportType
+
+val GLORY = intArrayOf(
+        Items.AMULET_OF_GLORY1, Items.AMULET_OF_GLORY2, Items.AMULET_OF_GLORY3,
+        Items.AMULET_OF_GLORY4, Items.AMULET_OF_GLORY5, Items.AMULET_OF_GLORY6
+)
+
+GLORY.forEach { glory ->
+    on_equipment_option(glory, option = "Edgeville") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 3086, z = 3500, height = 0)) //Coordinates of edgeville
+        }
+    }
+
+    on_equipment_option(glory, option = "Karamja") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 2917, z = 3176, height = 0)) //Coordinates of karamja
+        }
+    }
+
+    on_equipment_option(glory, option = "Draynor Village") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 3103, z = 3249, height = 0)) //Coordinates of draynor village
+        }
+    }
+
+    on_equipment_option(glory, option = "Al kharid") {
+        player.queue(TaskPriority.STRONG) {
+            teleport(Tile(x = 3292, z = 3165, height = 0)) //Coordinates of al kharid
+        }
+    }
+}
+
+fun QueueTask.teleport(tile : Tile) {
+    if (player.canTeleport()) {
+        if (player.hasEquipped(EquipmentType.AMULET, *GLORY)) {
+            world.spawn(AreaSound(tile, id = 200, radius = 10, volume = 1))
```

Comment:
The `AreaSound` should be a constant instead to avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).